devops/providers/azure-devops/templates/infrastructure/azure-pipeline-build-stage.yml

```diff
@@ -1,11 +1,13 @@
 parameters:
+  alwaysRun: 'false'
   environment: ''
   configurationMatrix: []
 
 jobs:
 
 - job: TemplateChangeDetection
   displayName: Determine CI Targets to Run
+  condition: ne(${{ parameters.alwaysRun }}, 'true')
```

Comment:
Can you help me understand why this needed here? Seems like we would always want to run the job and defer to the condition on line 26.

Reply:
Thinking we can save the execution time if we have a variable that will effectively override this check regardless. It's admittedly not much savings, but didn't seem necessary.